### DISCUSSION TITLE
What does ItemKey refer to in each tag type?

### DISCUSSION THREAD

**Discussion by Ferry-200** [Serial-ATA/lofty-rs#410] [Q&A] [answered]
Hello! Thanks for this great crate.

Recently I found it really convenient to get lyric by code:

```rust
        let tag = tagged_file.primary_tag().or(tagged_file.first_tag())?;
        return Some(tag.get(&ItemKey::Lyrics)?.value().text()?.to_string());
```

Because I can get lyric from music in different file format, such as mp3, flac, ogg, opus, wav, m4a, ...

Thus, I wonder what does ItemKey refer to in each tag type. Is it like this?

Id3v2: `USLT`, ~`SYLT`~
Vorbis Comment: `LYRICS`
Mp4Ilst: `©lyr`, ~`----:com.apple.iTunes:unsynced lyrics`~

**Comment by Serial-ATA** [comment, marked as answer]
Hello!

All of the `ItemKey` mappings are defined here: https://github.com/Serial-ATA/lofty-rs/blob/main/lofty/src/tag/item.rs.

`ItemKey::Lyrics` is for unsynced lyrics, so they map to:

* APE - `Lyrics`
* ID3v2 - `USLT`
* Mp4Ilst - `©lyr`
* Vorbis Comments - `LYRICS`
